backend.py

```diff
@@ -440,6 +441,22 @@ def performInstallation(answers, ui_package, interactive):
                                "latest dedicated ISO." %
                                (answers['linstor-version'], ', '.join(available_linstor_versions)))
 
+    # if upgrading a host with xapi, check we can upgrade it first
+    if answers['install-type'] == INSTALL_TYPE_REINSTALL and answers['xapi-version']:
+        available_xapi_versions = repository.listPackagesFromRepos(
+            main_repositories, 'xapi-core', '%{evr}')
+        if not available_xapi_versions:
+            raise RuntimeError("No xapi package found")
+        available_xapi_version = available_xapi_versions[0]
+        for v in available_xapi_versions[1:]:
+            if rpm.labelCompare(('pkg', 'arch', v), ('pkg', 'arch', available_xapi_version)) >= 0:
+                available_xapi_version = v
```

Comment:
behaviour when/if several versions are available is "latest wins", we may prefer to assert that we have a single version and avoid looping

Reply:
Indeed: `repoquery` needs `--show-duplicates` if you want more than one version. At least in the version that we're using.